File: Acume/src/main/java/com/guavus/acume/utility/ModifiedThreadFactory.java
package com.guavus.acume.utility;

import java.util.concurrent.ThreadFactory;
import java.util.concurrent.atomic.AtomicInteger;

public class ModifiedThreadFactory implements ThreadFactory {

    private static final AtomicInteger poolNumber = new AtomicInteger(1);
    private final ThreadGroup group;
    private final AtomicInteger threadNumber = new AtomicInteger(1);
    private final String namePrefix;

    
    ModifiedThreadFactory() {
        SecurityManager s = System.getSecurityManager();
        group = (s != null) ? s.getThreadGroup() :
                              Thread.currentThread().getThreadGroup();
        namePrefix = "pool-" +
                      poolNumber.getAndIncrement() +
                     "-thread-";
    }
    
    
    ModifiedThreadFactory(String threadprefix ) { 
        SecurityManager s = System.getSecurityManager();
        group = (s != null) ? s.getThreadGroup() :
                              Thread.currentThread().getThreadGroup();
        namePrefix = threadprefix + " - thread - ";
    }

    public Thread newThread(Runnable r) {
        Thread t = new Thread(group, r,
                              namePrefix + threadNumber.getAndIncrement(),
                              0);
        if (t.isDaemon())
            t.setDaemon(false);
        if (t.getPriority() != Thread.NORM_PRIORITY)
            t.setPriority(Thread.NORM_PRIORITY);
        return t;
    }
}

File: cache/src/main/scala/com/guavus/acume/cache/utility/ModifiedThreadFactory.scala
package com.guavus.acume.cache.utility

import java.util.concurrent.atomic.AtomicInteger
import java.util.concurrent.ThreadFactory

/**
 * @author archit.thakur
 * 
 * @param thd
 */
class ModifiedThreadFactory extends ThreadFactory {

  val poolNumber: AtomicInteger = new AtomicInteger(1);
  val threadNumber: AtomicInteger = new AtomicInteger(1);
  
  val s: Option[SecurityManager] = Option(System.getSecurityManager())
        
  val group = s match { 
    
    case Some(x) => x.getThreadGroup()              
    case None => Thread.currentThread().getThreadGroup();
  }
  var namePrefix = "pool-" + poolNumber.getAndIncrement() + "-thread-";
  
  def this(thd: String) = { 
    
    this()
    namePrefix = thd + " - thread - ";
  }
  
  def newThread(r: Runnable) = {
     
    val t = new Thread(group, r, namePrefix + threadNumber.getAndIncrement(), 0)
    if (t.isDaemon())
      t.setDaemon(false)
    if (t.getPriority() != Thread.NORM_PRIORITY)
      t.setPriority(Thread.NORM_PRIORITY)
    t
  }
}

